service-clients-gen/src/main/java/com/flipkart/poseidon/serviceclients/generator/ServiceGenerator.java

```diff
@@ -427,7 +468,9 @@ private JInvocation createRequest(ServiceIDL serviceIdl, JCodeModel jCodeModel,
                 builderInvocation = builderInvocation.invoke("setRequestObject").arg(JExpr.ref(requestObjectName));
             }
 
-            if (endPoint.getCommandName() != null && !endPoint.getCommandName().isEmpty()) {
+            if (endPoint.isIncludeMetaInfo()) {
+                builderInvocation = builderInvocation.invoke("setCommandName").arg(JExpr.ref(COMMAND_NAME_VAR_NAME));
```

Comment:
check whether command is to be dynamic, not everyone using metaInfo would need it

Reply:
Not needed. If includeMetaInfo is true, we are intialising COMMAND_NAME_VAR_NAME with either customCommandName Or getCommandName method